code/modules/events/minor/trader.dm

```diff
@@ -25,6 +25,13 @@
 #endif
 		var/docked_where = shuttle == "diner" ? "space diner" : "station";
 		command_alert("A merchant shuttle will dock with the [docked_where] shortly.", "Commerce and Customs Alert")
+		if(shuttle == "diner")
+			SEND_GLOBAL_SIGNAL(COMSIG_TRADER_DINER)
+		else
+			if(shuttle == "left")
+				SEND_GLOBAL_SIGNAL(COMSIG_TRADER_LEFT)
+			else
+				SEND_GLOBAL_SIGNAL(COMSIG_TRADER_RIGHT)
```

Comment:
switch would be more readable and extensible than elif chaining

Reply:
made this a proc, too :)